rfcs/0000-c-unwind-and-project-group-charter.md

```diff
@@ -6,146 +6,152 @@
-    * The "project group" term is newly introduced, but it corresponds
-      to a kind of working group -- one with the goal of fleshing out
-      a particular proposal or completing a project.
-    * We aim to specify how "C unwind" works on major platforms
-    * Goal is to enable Rust panics to propagate across native frames
-    * And perhaps to enable native exceptions to propagate across Rust frames
-    * But not to allow catching or throwing native exceptions from Rust code
+* Rust functions with the "C" abi will (abort on unwinding)[rust-lang/rust#52652].
+* The "C unwind" ABI will be introduced with unspecified behavior but the
+  explicit intention to permit Rust panics to propogate through native frames
+* Announce "project group" with the purpose of designing subsequent RFCs to
+  flesh out the details of the "C unwind" ABI; invite interested Rustaceans to
+  join
 
 # Motivation
 [motivation]: #motivation
 
-## We have no FFI ABI that permits unwinding, and we need one
+## We have no FFI ABI that permits unwinding, and we need one.
 
 Rust has long held that the `extern "C"` ABI does not permit
 unwinding. Therefore, attempts to unwind through a `extern "C"`
 barrier are considered [Undefined Behavior]. However, in practice, a
-number of projects have found that "things work fine" so long as
-certain conditions are met (for example, the frames that are to be
-unwound do not contain destructors).
+number of projects have found that "things work fine" in certain cases, such as
+when unwinding over C code compiled with Clang using the `-fexceptions`
+compiler option.
+
+> XXX I don't believe we've ever seen improper destructor behavior when
+> exposing `panic`s.
 
-We have made a number of efforts to "tighten up" the behavior around
-the C ABI and unwinding. For example, when users define a Rust
+We have made a number of efforts to define the behavior around
+the C ABI and unwinding to be safer. For example, when users define a Rust
 function that has the C ABI, we would like to make such functions
 abort if unwinding actually occurs at runtime. Unfortunately, these
 attempts have caused existing, working projects to break.
 
 The problem here is not that the existing projects break *per se*: they
-are relying on [Undefined Behavior] and that is to be expected as a
+are relying on [Undefined Behavior], so breakage is to be expected as a
 possibility. The problem is that there is no alternative available to
-them that could allow them to keep working (even if they are
+them that would allow them to keep working (even if they are
 continuing to rely on behavior that is not yet fully specified).
 
 ## Specific use cases
 
-We have seen a number of unwinding use cases in the wild. This section
+We have seen a number of cross-language unwinding use cases in the wild. This section
 details some of those use cases and their specific requirements.
 
-### Lucet: unwind past empty native frames
+### Lucet: unwind past foreign frames
```

Comment:
Discussion on changing "native" to "foreign" here: https://rust-lang.zulipchat.com/#narrow/stream/210922-wg-ffi-unwind/topic/Native.20frame.20definition/near/178713063